arweave/caip122.md

```diff
@@ -0,0 +1,93 @@
+---
+namespace-identifier: arweave-caip122
+title: Arweave Namespace - SIWx
+author: Rohit Pathare (@ropats16), Phil Billingsby (@pbillingsby), Dan MacDonald (@DanMacDonald)
+discussions-to: TBA
+status: Draft
+type: Standard
+created: 2022-09-01
+requires: ["CAIP-122", "CAIP-2", "CAIP-10"]
+---
+
+## CAIP-122
+
+For context, see the [CAIP-122](CAIP-122) specification.
+
+## Rationale
+
+Arweave signatures are 512 byte arrays which are `Base64URL` encoded for transmission over HTTP. This specification provides the signing algorithm to use, the `type` of the signing algorithm to identify it, and a method for signature creation and verification as required by [CAIP-122](CAIP-122).
+
+## Specification
+
+### Signing Algorithm
+
+Arweave wallets are RSA key pairs and use the private and public keys for signing and verifying signatures. Arweave wallet addresses are the SHA-256 hash of the RSA public key.
+
+### Signature Type
+
+Arweave signatures use the RSA-PSS signing algorithm to sign data and verify signatures.
```

Comment:
You need to specify a singature type string. See: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-74.md#signature-verification

Reply:
Do you mean something like Solana has specified on this line here?

https://github.com/ceramicnetwork/cacao/blob/d18d3f40bddd07d7a4ee5959f003c3416c9e1442/src/cacao.ts#L38

Reply:
Yes exactly.